legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabPatchApi.java

```diff
@@ -0,0 +1,249 @@
+            throw new LegendSDLCServerException("Failed to create patch release " + targetVersionId + " in project " + projectId);
+        }
+
+        return patchBranchToPatch(projectId, branch);
+    }
+
+    @Override
+    public List<Patch> getAllPatches(String projectId)
+    {
+        LegendSDLCServerException.validateNonNull(projectId, "projectId may not be null");
+        try
+        {
+            GitLabProjectId gitLabProjectId = parseProjectId(projectId);
+            String branchPrefix = getPatchReleaseBranchName("");
+            Pager<Branch> pager = getGitLabApi().getRepositoryApi().getBranches(gitLabProjectId.getGitLabId(), "^" + branchPrefix, ITEMS_PER_PAGE);
+            return PagerTools.stream(pager)
+                    .filter(branch -> (branch != null) && (branch.getName() != null) && branch.getName().startsWith(branchPrefix))
+                    .map(branch -> patchBranchToPatch(projectId, branch))
+                    .collect(PagerTools.listCollector(pager));
+        }
+        catch (Exception e)
+        {
+            throw buildException(e,
+                    () -> "User " + getCurrentUser() + " is not allowed to get patch release branches for project " + projectId,
+                    () -> "Unknown project: " + projectId,
+                    () -> "Error getting get patch release branches for project " + projectId);
+        }
+    }
+
+    private Patch fromPatchBranchName(String projectId, String branchName)
+    {
+        int index = branchName.lastIndexOf(BRANCH_DELIMITER);
+        String patchReleaseId = branchName.substring(index + 1, branchName.length());
+        return new Patch()
+        {
+            @Override
+            public String getProjectId()
+            {
+                return projectId;
+            }
+
+            @Override
+            public String getPatchReleaseVersion()
+            {
+                return patchReleaseId;
+            }
+        };
+    }
+
+    @Override
+    public void deletePatch(String projectId, String patchReleaseVersion)
+    {
+        LegendSDLCServerException.validateNonNull(projectId, "projectId may not be null");
+        LegendSDLCServerException.validateNonNull(patchReleaseVersion, "patchReleaseVersion may not be null");
+
+        GitLabProjectId gitLabProjectId = parseProjectId(projectId);
+        RepositoryApi repositoryApi = getGitLabApi().getRepositoryApi();
+
+        // Delete patch branch
+        boolean patchBranchDeleted;
+        try
+        {
+            patchBranchDeleted = GitLabApiTools.deleteBranchAndVerify(repositoryApi, gitLabProjectId.getGitLabId(), getPatchReleaseBranchName(patchReleaseVersion), 20, 1_000);
+            // close merge requests created for patch release branch
+            submitBackgroundRetryableTask(() -> closeMergeRequestsCreatedForPatchReleaseBranch(projectId, patchReleaseVersion), 5000L, "close merge requests created for branch " + getPatchReleaseBranchName(patchReleaseVersion));
```

Comment:
I think this will happen automatically when you delete the branch.

Reply:
No. It is not happening. It just says you cannot merge this MR. It doesn't close the MR.